### PR TITLE
HV-1116 Added support for javax.money validation

### PR DESCRIPTION
**Pull request by whiskeysierra**
This includes validators for `@DecimalMax`/`@DecimalMin` and `@Max`/`@Min`

### CODE REVIEW

File: engine/src/test/java/org/hibernate/validator/test/internal/xml/XmlMappingParserTest.java [hunk 1]
```diff
@@ -50,21 +51,23 @@ public void testAdditionalConstraintValidatorsGetAddedAndAreLastInList() {
 				DecimalMin.class
 		);
 
-		assertEquals( validators.size(), 2, "Wrong number of default validators" );
+		assertEquals( validators.size(), 3, "Wrong number of default validators" );
```
Comment (whiskeysierra) [inline engine/src/test/java/org/hibernate/validator/test/internal/xml/XmlMappingParserTest.java]: I feel like this test could be done in a less fragile way, but I didn't want to rewrite it without consent. My idea would be to just check whether the list is non-empty and the custom validator is the last, without also verifying the default validators explicitly.

**Comment by gunnarmorling** [comment]
Jenkins, add to whitelist.

File: engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/DecimalMinValidatorForCharSequence.java [hunk 2]
```diff
@@ -15,6 +15,9 @@
 import org.hibernate.validator.internal.util.logging.LoggerFactory;
 
 /**
+ * Check that the character sequence (e.g. string) being validated represents a number, and has a value
```
Comment (whiskeysierra) [inline engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/DecimalMinValidatorForCharSequence.java]: This class was missing a javadoc altogether, so I created one based on the `DecimalMaxValidatorForCharSequence`.

File: engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/DecimalMinValidatorForNumber.java [hunk 3]
```diff
@@ -16,7 +16,7 @@
 import org.hibernate.validator.internal.util.logging.LoggerFactory;
 
 /**
- * Check that the number being validated is less than or equal to the maximum
+ * Check that the number being validated is greater than or equal to the minimum
```
Comment (whiskeysierra) [inline engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/DecimalMinValidatorForNumber.java]: While looking at other validators and docs I realized that this was most likely a copy&paste error.

**Comment by AlexanderYastrebov** [comment]
Feels like it is only constrains the amount but not the currency. How one would validate amount boundaries and allowed currencies together?

**Comment by whiskeysierra** [comment]
> Feels like it is only constrains the amount but not the currency.

True. This limits the use of `@Max`/`@Min` on MonetaryAmount fields to either single currency situations and validation *around zero*, i.e. checking if a value is positive or negative.

> How one would validate amount boundaries and allowed currencies together?

For now, a custom constraint, I'd say.

**Comment by AlexanderYastrebov** [comment]
@whiskeysierra Got it - the idea is to add support for Money into Min/Max and DecimalMin/Max and not to implement full currency validation

**Comment by whiskeysierra** [comment]
> not to implement full currency validation

Exactly. If you have a good idea how that could look like, I'd be interested though.

**Comment by gunnarmorling** [comment]
> Exactly. If you have a good idea how that could look like, I'd be
interested though.

+1. What would be the requirements and how could a solution look like?

File: engine/pom.xml [hunk 4]
```diff
@@ -89,6 +89,12 @@
             <artifactId>paranamer</artifactId>
             <optional>true</optional>
         </dependency>
+        <dependency>
+            <groupId>javax.money</groupId>
+            <artifactId>money-api</artifactId>
+            <version>1.0.1</version>
```
Comment (gunnarmorling) [inline engine/pom.xml]: Could you add properties for these versions in the parent POM and add them to the `dependencyManagement` block in there?

File: engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/DecimalMinValidatorForCharSequence.java (hunk 2 above)
Comment (gunnarmorling) [inline engine/src/main/java/org/hibernate/validator/internal/constraintvalidators/bv/DecimalMinValidatorForCharSequence.java]: Nice!

**Comment by gunnarmorling** [comment]
Ah, and could you open an issue in the HV issue tracker (https://hibernate.atlassian.net/projects/HV/summary) and reword the commit message so it starts with the issue id? Thanks!

File: engine/src/test/java/org/hibernate/validator/test/internal/xml/XmlMappingParserTest.java (hunk 1 above)
Comment (gunnarmorling) [inline engine/src/test/java/org/hibernate/validator/test/internal/xml/XmlMappingParserTest.java]: Yes, that sounds good.

**Comment by AlexanderYastrebov** [comment]
> What would be the requirements and how could a solution look like?

I can imagine this additional constraints:

1. @MoneyMin/Max which constraints on amount and currency at the same time
2. @Currency which constraints only on set of currencies and could be mixed with @Min/Max/DecimalMin/DecimalMax

But this feels like a separate effort and goal of negative/positive check is achieved already with this one

**Comment by whiskeysierra** [comment]
@gunnarmorling I made the requested changes:

- version properties
- dependency management
- new issue
- commit message prefix
- mentioned supported data types in docs

Can you take a look?

**Comment by gunnarmorling** [comment]
Merged and rebased. Many thanks to you, @whiskeysierra and Lukas! I've added two more commits to add you two to copyright.txt and add a link to JSR 354 to the docs.

**Comment by gsmet** [comment]
Hmmm, the validators should have been added to the `hv` package, not the `bv` one, as they are not defined in the Bean Validation specification.

Could you create a PR to move them appropriately? (except if @gunnarmorling already moved them?)

**Comment by gsmet** [comment]
Oh and nice work, btw :).

**Comment by gsmet** [comment]
Oh, sorry I don't why I was redirected to this PR somehow. I thought it was a recent one.

Please ignore :).

**Pull request closed without merging**